app/uk/gov/hmrc/economiccrimelevyreturns/services/EmailService.scala

```diff
@@ -77,4 +77,30 @@ class EmailService @Inject() (emailConnector: EmailConnector, appConfig: AppConf
       throw e
     }
   }
+
+  def sendAmendReturnConfirmationEmail(eclReturn: EclReturn)(implicit
+    hc: HeaderCarrier,
+    messages: Messages
+  ): Future[Unit] = {
+    val obligationDetails: ObligationDetails = eclReturn.obligationDetails.getOrElse(
+      throw new IllegalStateException("No obligation details found in return data")
+    )
+    val dateSubmitted                        = ViewUtils.formatToday(translate = false)
+    val periodStartDate                      = ViewUtils.formatLocalDate(obligationDetails.inboundCorrespondenceFromDate)
+    val periodToDate                         = ViewUtils.formatLocalDate(obligationDetails.inboundCorrespondenceToDate)
+
+    (eclReturn.contactName, eclReturn.contactEmailAddress) match {
+      case (Some(name), Some(emailAddress)) =>
+        emailConnector.sendAmendReturnSubmittedEmail(
+          emailAddress,
+          AmendReturnSubmittedParameters(
+            name = name,
+            dateSubmitted = dateSubmitted,
+            periodStartDate = periodStartDate,
+            periodEndDate = periodToDate
+          )
+        )
+      case _                                => throw new IllegalStateException("Invalid contact details")
```

Comment:
We're breaking control flow by throwing exceptions. Can't we return an error and handle it in the controller?

Reply:
We have `ErrorHandler` which will automatically pick up this exception and apply standard handling. Should we handle it differently ?